tests/unit/test_publisher_backlist_service.py

```diff
@@ -0,0 +1,139 @@
+            "Last Modified": "2024-06-20T17:15:35.000Z",
+            "DRB_Record ID": "recRvPlHdLkM0KIem",
+            "Project Name (from Project)": ["Test Backlist"],
+            "Access type in DRB (from Access types)": [
+                "Limited access/login for read & download"
+            ],
+        },
+    }]
+                
+    def test_get_records(self, test_airtable_records, testService, mocker):
+        mock_records = mocker.patch('services.PublisherBacklistService.get_publisher_backlist_records', return_value=test_airtable_records)
+        mock_get = mocker.patch.object(testService.s3_manager, 'putObjectInBucket')
+        mock_get.return_value = {'ResponseMetadata': {'HTTPStatusCode': 200}}
+        test_get_records = testService.get_records()
+        assert len(test_get_records) == 2
+        mapped_full_access_record = test_get_records[0].record
+        full_access_manifest_link = mapped_full_access_record.has_part[0].split('|')
+        assert full_access_manifest_link[1] == 'https://regular-bucket.s3.amazonaws.com/manifests/publisher_backlist/Test Backlist/recPTBrKmZnsguK8M.json'
+        assert full_access_manifest_link[2] == 'Test Backlist'
+        assert full_access_manifest_link[3] == 'application/webpub+json'
+        assert full_access_manifest_link[4] == '{"catalog": false, "download": false, "reader": true, "embed": false}'
+        full_access_title_link = mapped_full_access_record.has_part[1].split('|')
+        assert full_access_title_link[1] == 'https://regular-bucket.s3.amazonaws.com/titles/publisher_backlist/Test Backlist/full_access.pdf'
+        assert full_access_title_link[2] == 'Test Backlist'
+        assert full_access_title_link[3] == 'application/pdf'
+        assert full_access_title_link[4] == '{"catalog": false, "download": true, "reader": false, "embed": false, "nypl_login": false}'
+        mapped_limited_access_record = test_get_records[1].record
+        limited_access_manifest_link = mapped_limited_access_record.has_part[0].split('|')
+        assert limited_access_manifest_link[1] == 'https://regular-bucket.s3.amazonaws.com/manifests/publisher_backlist/Test Backlist/recRvPlHdLkM0KIem.json'
+        assert limited_access_manifest_link[2] == 'Test Backlist'
+        assert limited_access_manifest_link[3] == 'application/webpub+json'
+        assert limited_access_manifest_link[4] == '{"catalog": false, "download": false, "reader": true, "embed": false, "fulfill_limited_access": false}'
+        limited_access_title_link = mapped_limited_access_record.has_part[1].split('|')
+        assert limited_access_title_link[1] == 'https://limited-bucket.s3.amazonaws.com/titles/publisher_backlist/Test Backlist/limited_access.pdf'
+        assert limited_access_title_link[2] == 'Test Backlist'
+        assert limited_access_title_link[3] == 'application/pdf'
+        assert limited_access_title_link[4] == '{"catalog": false, "download": true, "reader": false, "embed": false, "nypl_login": true}'
```

Comment:
Can we simplify the assertions here? It may be helpful to write helper functions that build the links.